File: app/src/main/java/ru/SnowVolf/pcompiler/patch/PatchCollection.java
package ru.SnowVolf.pcompiler.patch;


/**
 * Created by Snow Volf on 23.10.2017, 21:20
 */

public class PatchCollection {
    private static PatchArray collection = null;

    public static PatchArray getCollection(){
        if (collection == null){
            collection = new PatchArray();
        }
        return collection;
    }
}

File: app/src/main/java/ru/SnowVolf/pcompiler/patch/PatchCollection.kt
package ru.SnowVolf.pcompiler.patch


/**
 * Created by Snow Volf on 23.10.2017, 21:20
 */

object PatchCollection {
    private var collection: PatchArray? = null

    fun getCollection(): PatchArray {
        if (collection == null) {
            collection = PatchArray()
        }
        return collection!!
    }
}
